andreeapetcu/assignment3/TrieNode.java

```diff
@@ -0,0 +1,208 @@
+import java.util.ArrayList;
+
+/**
+ * Modified implementation of a trie node such that it contains the position (row and column index)
+ * of a character in the grid
+ *
+ * ASSUMPTIONS : ASCII character encoding, lower case letters only
+ */
+
+public class TrieNode {
+  // the number of letters in the alphabet
+  private final static int LETTERS = 25;
+
+  private char value;
+  private TrieNode parent;
+  // the row index of the node in the grid
+  private int row;
```

Comment:
I wouldn't have introduced the idea of row and column into your Trie. This seems like a mistake. But I'll read the rest of the code.